source/MaterialXCore/Document.cpp

```diff
@@ -266,7 +266,7 @@ NodeDefPtr Document::addNodeDefFromGraph(const NodeGraphPtr nodeGraph, const str
     return nodeDef;
 }
 
-void Document::importLibrary(const ConstDocumentPtr& library)
+void Document::importLibrary(const ConstDocumentPtr& library, bool errorOnDuplicates)
```

Comment:
This error condition used to reside within the read options at some point in the past. It might be better to put it there instead ? AFAIK it still resides in the Autodesk fork so adding a ping for @ashwinbhat for any feedback.

Reply:
What I was actually trying to catch here was the same node definition being defined in two different library files, so I believe this wouldn't be possible to catch with a read options flag.  As I understood things the read options only apply to when a single file is loaded.

Adding the check here, allows clashes from different files to be validates.  I understand that there are certainly use-cases for not wanting this to be an error, hence the optional guard.